### DISCUSSION TITLE
2 Question about queues

### DISCUSSION THREAD

**Discussion by trsh** [OptimalBits/bull#2333] [Q&A]
1) Let's say I create a queue named 'xxx'. Then for some reason my nodejs server crashes and restarts. Then, again, I create queue with same name 'xxx'. What will happen? Will it be an entirely new (empty) queue or will it get back Jobs hanging in redis?

2) How do I clean up garbage in redis created by bull? In those cases where, for example again, the node js server crashes and restarts. And I don't want any of saved queues and jobs.

**Comment by manast** [comment]
> 1. Let's say I create a queue named 'xxx'. Then for some reason my nodejs server crashes and restarts. Then, again, I create queue with same name 'xxx'. What will happen? Will it be an entirely new (empty) queue or will it get back Jobs hanging in redis?

When you create a new instance of a Queue you do not create a queue, just connects to it so that you can add or consume jobs.

> 2. How do I clean up garbage in redis created by bull? In those cases where, for example again, the node js server crashes and restarts. And I don't want any of saved queues and jobs.

https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queueobliterate

**Reply by trsh** [reply]
> When you create a new instance of a Queue you do not create a queue, just connects to it so that you can add or consume jobs.

So after my node js server restart, if I create a queue instance with same name, it will pick up existing Jobs save from Redis store?

> https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queueobliterate

Ok. So I can use this method to flush the Queue. But how do I retrieve all existing queues from redis, for example, after nodejs server restart?

**Reply by trsh** [reply]
I want to make sure to retire old unused queues.
